front/lib/api/assistant/conversation.ts

```diff
@@ -143,18 +149,22 @@ export async function updateConversation(
     throw new Error(`Conversation ${conversationId} not found`);
   }
 
+  if (!(await canAccessConversation(auth, conversation))) {
+    throw new Error("Conversation access denied");
+  }
+
   await conversation.update({
     title: title,
     visibility: visibility,
   });
 
-  const c = await getConversation(auth, conversationId);
+  const cRes = await getConversation(auth, conversationId);
 
-  if (!c) {
-    throw new Error(`Conversation ${conversationId} not found`);
+  if (cRes.isErr()) {
+    throw cRes.error;
```

Comment:
We used to throw as this error was unexpected but now that this is a more versatile error do we really want to throw?

Reply:
IMO, both cases are quite unexpected; update of a forbidden convo is impossible via the UI and adversarial via APi so I thought it was fine to throw

Not strong on this, so ok to evolve